### ISSUE TITLE
[Bug] Windows build failing on multiply defined symbol "load_plugin"

### ISSUE THREAD

**Issue by andrew-otiv** [eclipse-zenoh/zenoh-plugin-mqtt#19]
### Describe the bug

I hit the following linker error:

```
   Compiling zenoh-bridge-mqtt v0.11.0-dev (C:\Users\doruk\zenoh-plugin-mqtt\zenoh-bridge-mqtt)
error: linking with `link.exe` failed: exit code: 1169

...  = note: libzenoh_plugin_mqtt.rlib(zenoh_plugin_mqtt.2ih4zof794c45w3y.rcgu.o) : error LNK2005: load_plugin already defined in libzenoh_plugin_rest-a198eafe84970acb.rlib(zenoh_plugin_rest-a198eafe84970acb.zenoh_plugin_rest.1c88527b5be1b9e4-cgu.04.rcgu.o)
          C:\Users\doruk\zenoh-plugin-mqtt\target\debug\deps\zenoh_bridge_mqtt.exe : fatal error LNK1169: one or more multiply defined symbols found

error: could not compile `zenoh-bridge-mqtt` (bin "zenoh-bridge-mqtt") due to previous error
```

### To reproduce

cargo build

### System info

Microsoft Windows 11 Home

rustc 1.72.0 (5680fa18f 2023-08-23)

zenoh-plugin-mqtt
commit 10a8bda49926ef68d50d2e63caf8c1293421285b (HEAD -> master, tag: 0.11.0-dev, origin/master, origin/HEAD)
Date:   Wed Oct 4 10:38:52 2023 +0200

**Comment by JEnoch** [comment]
**TL;DR:**
This error occurs on Windows or Linux when building both plugin (`zenoh-plugin-mqtt`) and standalone executable (`zenoh-bridge-mqtt`) at the same time with the `cargo build` command.
You must build each independently as per [README instructions](https://github.com/eclipse-zenoh/zenoh-plugin-mqtt#how-to-build-it):
  - `cargo build --release -p zenoh-plugin-mqtt`
  - `cargo build --release -p zenoh-bridge-mqtt`

I will improve the README to emphasis this.

**Root cause of the issue**:
To be dynamically loaded by the Zenoh router (`zenohd`), a Zenoh plugin must defines a `load_plugin()` operation in C representation and non-mangled.
In the `zenoh-bridge-mqtt` we rather statically link 2 plugins (`zenoh-plugin-mqtt` and `zenoh-plugin-rest`). If those plugins both define the same non-mangled `load_plugin()` operation,`ld` reports a conflict because of multiple definitions. 
Therefore, in all plugins `Cargo.toml`, we define a `no_mangle` feature (active by default) that can can be deactivated when statically linking several plugins in a same bin. `zenoh-bridge-mqtt/Cargo.toml` well [deactivates this feature](https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/blob/f1799d1afb0b4a34e089a9485397bfb8e7eaf0fe/zenoh-bridge-mqtt/Cargo.toml#L35). But unfortunately, when building both `zenoh-plugin-mqtt` and `zenoh-bridge-mqtt` together, Cargo build `zenoh-plugin-mqtt` only once and with this `no_mangle` feature enabled.

**Comment by andrew-otiv** [comment]
Thanks! I downloaded the artifact, and zenoh-bridge-mqtt.exe runs cleanly on my machine with no parameters. (I haven't tested if it relays messages yet).
